js/rank_states_utils.js

```diff
@@ -3,3 +3,142 @@ var svgStates = d3.select("#rank-states-interactive")
+
+var dragStates = ["ID","OK","MI"];
+
+// add states
+  svgStates.select('#ranked-states-moved')
+    .selectAll( 'use' )
+    .data(movedStates)
+    .enter()
+    .append('use')
+    .style('stroke-dasharray',"10, 10")
+    .classed('moved-state', true)
+    .attr('xlink:href', function(d) {
+      return '#'+ d +'-pattern';
+    });
+  
+  svgStates.select('#ranked-states-dragable')
+    .selectAll( 'use' )
+    .data(dragStates)
+    .enter()
+    .append('use')
+    .classed('draggable','true') 
+    .style("fill",categoryToColor("total"))
+    .style("stroke-dasharray",null)
+    .attr('id', function(d) {
+      return d + '-rank';
+    })
+    .attr('xlink:href', function(d) {
+      return '#'+ d +'-pattern';
+    })
+    .datum({x: 0, y: 0})
+    .call(d3.drag()
+      .on("drag", dragging)
+      .on("end", dragdone));
+    
+  var bardata = [{"wu":80, "abrv":'OK', "open": true}, {"wu":140, "abrv":'WI', "open":false}, {"wu":180, "abrv":'MI', "open":true}, 
+    {"wu":210, "abrv":'ID', "open":true}, {"wu":260, "abrv":'TX', "open":false}, {"wu":310, "abrv":'CA', "open":false}];
+  
+  var barGroups = svgStates.select('#ranked-states-bars')
+    .selectAll('g')
+    .data(bardata)
+    .enter()
+    .append('g')
+    .attr('transform', function(d, i){
+      return 'translate(' + (500 + i *30) + "," + (280-d.wu) + ")";
+    });
+  
+  barGroups.append('text')
+    .attr('y', function(d){
+      return d.wu;
+    })
+    .classed('bar-name',true)
+    .classed('open-bar-name', function(d){
+      return d.open;
+    })
+    .attr('dy',"1em")
+    .text(function(d){
+      return d.abrv;
+    });
+  
+  barGroups.append('rect')
+    .attr('height', function(d){
+      return d.wu;
+    })
+    .attr('width','20')
+    .style("stroke-dasharray","4, 2")
+    .classed('open-rank-bar', function(d){
+      return d.open;
+    })
+    .classed('closed-rank-bar', function(d){
+      return !d.open;
+    })
+    .style('fill', function(d){
+      if (!d.open){
+        return categoryToColor("total");
+      } else {
+        null;
+      }
+    })
+    .on('mouseover', function(d){
+      if (d.open){
+        d3.select(this)
+          .style("stroke-dasharray",null)
+          .classed('chosen-rank-bar',true)
+          .style("fill", categoryToColor("total"));
+      } else {
+        null;
+      }
+      })
+    .on('mouseout', function(d){
+      if (d.open){
+        d3.select(this)
+          .style("stroke-dasharray","4, 2")
+          .classed('chosen-rank-bar',false);
+        } else {
+          null;
+        }
+      });
+  
+    function dragging(d) {
+      d3.select(this).attr("transform", "translate(" + (d.x = d3.event.x) + "," + (d.y = d3.event.y) + ")");
```

Comment:
I don't think you need `d.x =` and `d.y =` since you are specifying a translate value

Reply:
Gotcha - these d.x and d.y parts are holdovers from when I was letting the state be dropped and picked back up, and we needed to capture the origin of the event. It didn't work very well anyhow..